src/Castle.Core/DynamicProxy/Tokens/InvocationMethods.cs

```diff
@@ -89,7 +89,7 @@ public static class InvocationMethods
 		public static readonly MethodInfo SetReturnValue =
 			typeof(AbstractInvocation).GetMethod("set_ReturnValue");
 
-		public static readonly FieldInfo Target =
+		public static readonly FieldInfo CompositionInvocationTarget =
```

Comment:
Just realised these fields are public, so technically breaking. I'm thinking if anyone is using them directly rather than via one of the supported interfaces (e.g. `IProxyTargetAccessor`) then bad luck to them, and we should make the new ones internal.

Reply:
True! 🤔 Since we're already about to break something, how about we deprecate *all* those `…Methods` types in the `Tokens` folder? I could submit a separate PR for that to go before this one.

Reply:
What about just keeping back compat like this then:

```csharp
public static readonly FieldInfo CompositionInvocationTarget = typeof...;
public static readonly FieldInfo Target = CompositionInvocationTarget;
```

And we worry about deprecating them with the other task.

Reply:
Done.

Reply:
> Shall I still submit a PR just for deprecating Tokens/, or would you prefer a larger PR that deprecates a lot of things all at once?

Let's just leave it to the full API review, one big release with all the deprecations should be easier to manage.